source_control/github_key.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+module: github_key
+short_description: Manage GitHub access keys.
+description:
+    - Creates, removes, or updates GitHub access keys.
+version_added: "2.0"
+options:
+  token:
+    description:
+      - GitHub Access Token with permission to list and create public keys.
+    required: true
+  name:
+    description:
+      - SSH key name
+    required: true
+  pubkey:
+    description:
+      - SSH public key value. Required when state=present.
+    required: false
+  state:
+    description:
+      - Whether to remove a key, ensure that it exists, or update its value.
+    choices: ['present', 'absent']
+    default: 'present'
+    required: false
+  force:
+    description:
+      - The default is C(yes), which will replace the existing remote key
+        if it's different than C(pubkey). If C(no), the key will only be
+        set if no key with the given C(name) exists.
+    required: false
+    choices: ['yes', 'no']
+    default: 'yes'
+
+author: Robert Estelle (@erydo)
+'''
+
+EXAMPLES = '''
+- name: Read SSH public key to authorize
+  shell: cat /home/foo/.ssh/id_rsa.pub
+  register: ssh_pub_key
+
+- name: Authorize key with GitHub
+  local_action:
+    module: github_key
+    name: 'Access Key for Some Machine'
+    token: '{{github_access_token}}'
+    pubkey: '{{ssh_pub_key.stdout}}'
```

Comment:
This module is not intended to register a local machine. Doing so would require saving a privileged github access token to that machine first, and there's no or little benefit to going through that process versus just adding your local key by hand.

The module is intended for registering remote systems: servers, build machines and so on, and I think that showing the local example might be confusing. A newcomer may see that example and fail to make the distinction, and perhaps even not realize that they should have separate keys on different machines.